scipy/stats/_distribution_infrastructure.py

```diff
@@ -0,0 +1,5082 @@
+
+        Evaluate the log-PDF at the desired argument:
+
+        >>> X.logpdf(0.5)
+        -0.6931471805599453
+        >>> np.allclose(X.logpdf(0.5), np.log(X.pdf(0.5)))
+        True
+
+        """
+        return self._logpdf_dispatch(x, method=method, **self._parameters)
+
+    @_dispatch
+    def _logpdf_dispatch(self, x, *, method=None, **kwargs):
+        if self._overrides('_logpdf_formula'):
+            method = self._logpdf_formula
+        elif self.tol is _null:  # ensure that developers override _logpdf
+            method = self._logpdf_logexp
+        return method
+
+    def _logpdf_formula(self, x, **kwargs):
+        raise NotImplementedError(self._not_implemented)
+
+    def _logpdf_logexp(self, x, **kwargs):
+        return np.log(self._pdf_dispatch(x, **kwargs))
+
+    @_set_invalid_nan
+    def pdf(self, x, *, method=None):
+        r"""Probability density function
+
+        The probability density function :math:`f(x)` is the probability *per
+        unit interval* of :math:`x` that the random variable takes on the value
+        :math:`x`. Mathematically, it is defined as the derivative of the
+        cumulative distribution function `F(x)`:
+
+        .. math::
+
+            f(x) = \frac{d}{dx} F(x)
+
+        ``pdf`` accepts `x` for :math:`x`.
+
+        Parameters
+        ----------
+        x : Array
+            The value of the random variable for which the PDF is to be
+            evaluated.
+        method : {None, 'formula', 'logexp'}
+            The strategy used to evaluate the PDF. By default (``None``), the
+            infrastructure chooses between the following options.
+
+            - ``'formula'``: use a formula for the PDF itself
+            - ``'logexp'``: evaluate the logarithm of the PDF directly and
+                            exponentiate
+
+            Not all `method` options are available for all distributions.
+            If the selected `method` is not available, a ``NotImplementedError``
+            will be raised.
+
+        Returns
+        -------
+        out : Array
+            The PDF evaluated at the argument `x`.
+
+        See Also
+        --------
+        ContinuousDistribution.cdf
+        ContinuousDistribution.logpdf
+
+        Notes
+        -----
+        Suppose a probability distribution has support :math:`[l, r]`.
+        By definition of the support, the PDF evaluates to its minimum value
+        of :math:`0` outside the support; i.e. for :math:`x < l` or
+        :math:`x > r`. The maximum of the PDF may be less than or greater than
+        :math:`1`; since the valus is a probability *density*, only its integral
+        over the support must equal :math:`1`.
+
+        Examples
+        --------
+        Instantiate a distribution with the desired parameters:
+
+        >>> from scipy import stats
+        >>> X = stats.Uniform(a=-1., b=1.)
+
+        Evaluate the probability density function at the desired argument:
```

Comment:
I suggest to add a bit more background: For a uniform distribution, the pdf is constant over the support. Its value is `1/(b-a)`: 
(evaluate the pdf at a few points)

Reply:
> or a uniform distribution, the pdf is constant over the support. Its value is 1/(b-a)

I used to think that API reference documentation should include this sort of information. For example, see the documentation of [`jarque_bera`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.jarque_bera.html) and several other hypothesis tests that go above and beyond in explaining the utility of the test and the meaning of the output. 

I no longer think that is appropriate for API documentation, so we are in the process of moving all that over to tutorials. Here, I think the example just needs to show how to call the `pdf` function, not expound upon the meaning of the uniform distribution in particular. 

> (evaluate the pdf at a few points)

We could do that in *every* example, or we could include a note about vectorization/broadcasting in the tutorial. I would prefer the latter.